### PR TITLE
Make jest react-native preset to use typescript files

### PR DESCRIPTION
**Pull request by dalcib**
Metro and react-native now have first-class support to Typescript with Babel 7, but react-native Jest preset needs specific settings to work with .ts and .tsx files.
This PR updates react-native Jest preset to use Typescript out of the box.

Test Plan:
----------
* [x]  Check yarn run lint
* [x]  Check yarn run prettier
* [x]  Check yarn run test
There is a test to the hasteImpl file.

Changelog:
----------

[GENERAL] [CHANGE] - Add Typescript to Jest react-native preset.

### CODE REVIEW

File: jest/hasteImpl.js [hunk 1]
```diff
@@ -57,8 +57,8 @@ const WHITELISTED_PREFIXES /*: Array<string> */ = [
 const NAME_REDUCERS /*: Array<[RegExp, string]> */ = [
   // extract basename
   [/^(?:.*[\\\/])?([a-zA-Z0-9$_.-]+)$/, '$1'],
-  // strip .js/.js.flow suffix
-  [/^(.*)\.js(\.flow)?$/, '$1'],
+  // strip .js/.ts/.tsx/.js.flow suffix
+  [/^(.*)\.(js|ts|tsx)(\.flow)?$/, '$1'],
```
Comment (dulmandakh) [inline jest/hasteImpl.js]: this regex will match ts.flow and tsx.flow

**Comment by TheSavior** [comment]
cc @rafeca in case there are additional files we need to be aware of.

File: jest/hasteImpl.js [hunk 2]
```diff
@@ -57,8 +57,8 @@ const WHITELISTED_PREFIXES /*: Array<string> */ = [
 const NAME_REDUCERS /*: Array<[RegExp, string]> */ = [
   // extract basename
   [/^(?:.*[\\\/])?([a-zA-Z0-9$_.-]+)$/, '$1'],
-  // strip .js/.js.flow suffix
-  [/^(.*)\.js(\.flow)?$/, '$1'],
+  // strip .js/.ts/.tsx/.js.flow suffix
```
Comment (cpojer) [inline jest/hasteImpl.js]: I don't think the changes to this file are necessary, are they? This only affects the react-native repo itself.

Comment (dalcib) [reply jest/hasteImpl.js]: @cpojer , you are right. I fixed it in https://github.com/facebook/react-native/pull/22217/commits/62551c2162e60b1a63c037089ae59b833602ebf0

**Pull request closed without merging**